htscodecs/rANS_static4x16pr.c

```diff
@@ -842,26 +846,28 @@ void rans_set_cpu(int opts) {
 #  define UNUSED
 #endif
 
-static inline
-unsigned char *(*rans_enc_func(int do_simd, int order))
-    (unsigned char *in,
-     unsigned int in_size,
-     unsigned char *out,
-     unsigned int *out_size) {
-    if (!do_simd) { // SIMD disabled
-        return order & 1
-            ? rans_compress_O1_4x16
-            : rans_compress_O0_4x16;
-    }
+// CPU detection is performed once.  NB this has an assumption that we're
+// not migrating between processes with different instruction stes, but
+// to date the only systems I know of that support this don't have different
+// capabilities (that we use) per core.
+#ifndef NO_THREADS
+static pthread_once_t rans_cpu_once = PTHREAD_ONCE_INIT;
+#endif
+
+static int have_ssse3   UNUSED = 0;
+static int have_sse4_1  UNUSED = 0;
+static int have_popcnt  UNUSED = 0;
+static int have_avx2    UNUSED = 0;
+static int have_avx512f UNUSED = 0;
+int is_amd              UNUSED = 0;
```

Comment:
Should this be `static` as well as the others?

Reply:
Yes, I thought I'd done so.  Maybe I changed it briefly for testing and forgot to put it back.

Good spot.